src/physics/em/detail/BetheHeitler.cu

```diff
@@ -53,9 +54,10 @@ bethe_heitler_interact_kernel(const BetheHeitlerPointers                bh,
                              particle.particle_id(),
                              material.material_id(),
                              tid);
+    SimTrackView     sim(model.states.sim, tid);
 
     // This interaction only applies if the Bethe-Heitler model was selected
-    if (physics.model_id() != bh.model_id)
+    if (physics.model_id() != bh.model_id || !sim.alive())
```

Comment:
This is for when there are more tracks than primaries. While all tracks initially have the sim state initialized with `alive = false`, we don't do the same for the physics state and model ID -- maybe we should instead of adding this check?

For active tracks you're right. If a track escapes the geometry it's killed, and it will have an invalid `ModelId` in that case (as in the case where it crosses a boundary) because the mfp will be nonzero. Stopped particles that can't undergo a discrete interaction are killed, and those will have an invalid `ModelId` since they have zero macro xs. Particles that are absorbed are killed post-interaction, so not relevant here.

Reply:
Good point, or maybe we should put this into the `select_model` bit: instead of `return`ing if `!sim.alive()`, clear the model ID so it's not selected.

Reply:
I think we would need to set `model_id = ModelId{}` for all tracks during initialization since we launch the interaction kernels with the max number of tracks and should exit if the track is empty.

Reply:
Is initialization the right place? Is that going to apply to empty slots as the cascade dies out, in addition to the very first one? I was thinking in the `along_and_post_step_kernel` would work, since that's before the interaction kernels (where the model ID gets set for the active tracks).

Reply:
Setting it at initialization should be alright.. the physics state is cleared at the end of the post-processing kernel if a discrete interaction occurred, and new tracks from primaries/secondaries are initialized with an empty `ModelId`. But you're right, I think clearing it in `along_and_post_step_kernel` would also work and would be easier.